views/src/js/posts_list.js

```diff
@@ -0,0 +1,100 @@
+function ListItemHeader(props) {
+    return <h3>{props.post_header}</h3>;
+}
+
+function ListItemBody(props) {
+    const post = Object.entries(props.post);
+
+    return (
+        <ul>
+            {
+                post.map(post_field => <li key={post_field[0]}><b>{post_field[0] + ": "}</b>{post_field[1]}</li>)
+            }
+        </ul>
+    );
+}
+
+function ListItem(props) {
+    const post = props.post;
+
+    return (
+    <div>
+        <div id="list-item-header">
+            {
+                <ListItemHeader key={post.unique_id} post_header={post.post_category}/>
+            }
+        </div>
+        <div id="list-item-body">
+            {
+                <ListItemBody key={post.unique_id} post={post}/>
+            }
+        </div>
+    </div>
+    );
+}
+
+function ActionLink(props) {
+    const value = props.value;
+    const operation = props.operation;
+
+    function handleClick() {
+        const page = document.getElementById("current_page");
```

Comment:
page number should be a state which you update clicking pagination. all pagination fields should be a state, which updates from interface or by api data (pageCount, total, etc..) try to avoid DOM methods. it decrease react performance